pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/metrics/ManagedCursorMetrics.java

```diff
@@ -77,8 +85,32 @@ public ManagedCursorMetrics(PulsarService pulsar) {
                 metrics.put("brk_ml_cursor_persistLedgerErrors", cStats.getPersistLedgerErrors());
                 metrics.put("brk_ml_cursor_persistZookeeperSucceed", cStats.getPersistZookeeperSucceed());
                 metrics.put("brk_ml_cursor_persistZookeeperErrors", cStats.getPersistZookeeperErrors());
+                metrics.put("brk_ml_cursor_writeLedgerSize", cStats.getWriteCursorLedgerSize());
+                metrics.put("brk_ml_cursor_writeLedgerLogicalSize", cStats.getWriteCursorLedgerLogicalSize());
+                metrics.put("brk_ml_cursor_readLedgerSize", cStats.getReadCursorLedgerSize());
                 metricsCollection.add(metrics);
+
+                nsTotalNonContiguousDeletedMessagesRange += cursor.getTotalNonContiguousDeletedMessagesRange();
+                nsPersistLedgerSucceed += cStats.getPersistLedgerSucceed();
+                nsPersistLedgerErrors += cStats.getPersistLedgerErrors();
+                nsPersistZookeeperSucceed += cStats.getPersistZookeeperSucceed();
+                nsPersistZookeeperErrors += cStats.getPersistZookeeperErrors();
+                nsWriteCursorLedgerSize += cStats.getWriteCursorLedgerSize();
+                nsWriteCursorLedgerLogicalSize += cStats.getWriteCursorLedgerLogicalSize();
+                nsReadCursorLedgerSize += cStats.getReadCursorLedgerSize();
             }
+
+            Metrics metrics = createMetricsByDimension(namespace);
+            metrics.put("brk_ml_cursor_nonContiguousDeletedMessagesRange",
+                    nsTotalNonContiguousDeletedMessagesRange);
+            metrics.put("brk_ml_cursor_persistLedgerSucceed", nsPersistLedgerSucceed);
+            metrics.put("brk_ml_cursor_persistLedgerErrors", nsPersistLedgerErrors);
+            metrics.put("brk_ml_cursor_persistZookeeperSucceed", nsPersistZookeeperSucceed);
+            metrics.put("brk_ml_cursor_persistZookeeperErrors", nsPersistZookeeperErrors);
+            metrics.put("brk_ml_cursor_writeLedgerSize", nsWriteCursorLedgerSize);
+            metrics.put("brk_ml_cursor_writeLedgerLogicalSize", nsWriteCursorLedgerLogicalSize);
+            metrics.put("brk_ml_cursor_readLedgerSize", nsReadCursorLedgerSize);
+            metricsCollection.add(metrics);
```

Comment:
What is the purpose of adding these metrics with a single dimension?

Coming from a prometheus background (I'm not familiar with other metrics providers), these single dimension metrics seem like an anti-pattern. These metrics are already provided in the above metrics with the `ledger_name` and `cursor_name` dimensions. A user can easily get the count at the namespace level by just aggregating on the `namespace` dimension. Further, if they were to aggregate on all metrics named `brk_ml_cursor_persistLedgerSucceed`, they would get double the actual value of the metrics, which is problematic.

Reply:
nice catch @michaeljmarshall I think we should only expose the namespace dimension if exposeManagedCursorMetricsInPrometheus=false

Reply:
Thanks @michaeljmarshall  and @codelipenghui .
I will remove this later.